src/main/java/io/stargate/sgv2/jsonapi/service/operation/tasks/TaskGroup.java

```diff
@@ -0,0 +1,113 @@
+package io.stargate.sgv2.jsonapi.service.operation.tasks;
+
+import io.stargate.sgv2.jsonapi.service.cqldriver.executor.SchemaObject;
+import java.util.*;
+
+/**
+ * Ordered container of {@link Task} to be processed as a group.
+ *
+ * <p>The container can be configured to process the tasks sequentially or in parallel (default),
+ * and other config that is needed about how to process the tasks as a group should live here (e.g.
+ * if there is a delay between tasks, or if the container should fail fast).
+ *
+ * <p>TODO: aaron feb 4 - stop inheriting array list, just expose methods needed
+ *
+ * @param <SchemaT> Schema object type
+ */
+public class TaskGroup<TaskT extends Task<SchemaT>, SchemaT extends SchemaObject>
```

Comment:
Are these type parameters needed? As in, do TaskGroups need to be typed by `Task` type (and thereby `SchemaObject` type?)

Reply:
1) the generics got out of control, there is plenty of room for making this simple. 
2) I *think* it needs the generics because we need to know the Task type, so when it comes times to build the results we have the TaskT to do things like get the results for an insert or read task.